specific_use_cases/DFU_work_in_progress/INTRO_annual_consumption_model.ipynb

```diff
@@ -129,253 +92,192 @@
-   "id": "1900ddce-eed5-4ed4-83a0-63964df4b0ac",
-   "metadata": {
-    "tags": []
-   },
-   "source": [
-    "### 2a) Read in a csv file of the station coordinates \n",
-    "Make sure the filepath to the csv file matches the correct location on your computer. This file will be read into the notebook as a pandas DataFrame object."
+    "### 1b) Load the data into memory\n",
+    "This may take some time, because the data has to be loaded into memory and then subsetted to get the closest grid cell. All computations we've done before this step are actually computed in this step; before, we just see a preview of the data. Because of this, **we recommend running this notebook in the Analytics Engine's Jupyter Hub, which provides additional computational resources that greatly speed up this step.**"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "786c2ff2-b12e-4b31-af79-4648cd348e44",
+   "id": "82c96c9d-14cd-425e-8d9d-ad1b178e05c3",
    "metadata": {},
    "outputs": [],
    "source": [
-    "stations_df = pd.read_csv(\"data/CEC_Forecast_Weather Stations_California.csv\", index_col=\"STATION\")\n",
-    "stations_df.head(5) # Display the first 5 rows "
-   ]
-  },
-  {
-   "cell_type": "markdown",
-   "id": "1d4cd9b4-6df8-499a-b7d9-c45c782b3b87",
-   "metadata": {},
-   "source": [
-    "### 2b) Grab the closest grid cell to the weather station\n",
-    "To demonstrate this process, we'll use the Sacramento Executive Airport weather station."
+    "%%time\n",
+    "data_closest_gridcell = app.load(data_closest_gridcell)"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "9486019c-c1f7-4ac9-b63e-e2234de5dc00",
+   "id": "4c193431-34b3-4e39-8c84-f465d1c0442d",
    "metadata": {},
    "outputs": [],
    "source": [
-    "station_name = \"SACRAMENTO EXECUTIVE AIRPORT\"\n",
-    "one_station = stations_df.loc[station_name]"
+    "data_closest_gridcell"
    ]
   },
   {
    "cell_type": "markdown",
-   "id": "a16bc16c-d454-47f8-81d7-1f360e6c46df",
+   "id": "a68132d7-e91c-4555-ab0d-13af7d61e1b7",
    "metadata": {},
    "source": [
-    "Next, we need to convert the lat/lon coordinate pair to the model's projection coordinates. We can easily do this using the built in helper function in climakitae: `get_closest_gridcell`. For more information on this function, you can call `help(get_closest_gridcell)` or look in the climakitae.utils module for the actual code that performs the computation."
+    "### 1c) Output final data product as a csv file\n",
+    "We'll drop all unneeded coordinates and convert our xarray Dataset to a pandas Dataframe, allowing us to easily output the final data product to a csv file. In the output table, the first column is the time in units of UTC, and the second column are the various global climate models (which can be filtered in excel or in python code in the notebook). The other columns are the variables selected at the beginning of the notebook."
```

Comment:
There are some extra unneeded coordinates that do not appear (any longer) that it can't find to drop. And the 'station_name' is not recognized for replacing ' ' with '_'

Reply:
(similar thing happens in 3c also)

Reply:
(and 4g)